r/tools/nixlibs.R

```diff
@@ -539,10 +562,25 @@ build_libarrow <- function(src_dir, dst_dir) {
     env_var_list <- c(env_var_list, ARROW_DEPENDENCY_SOURCE = "BUNDLED")
   }
 
+  # On macOS, if not otherwise set, let's override Boost_SOURCE to be bundled
+  if (on_macos) {
+    deps_to_bundle <- c("Boost", "lz4")
+    for (dep_to_bundle in deps_to_bundle) {
+      env_var <- paste0(dep_to_bundle, "_SOURCE")
+      if (Sys.getenv(env_var) == "") {
+        # TODO: env_var_list gets checked for caps, so we need to do that, but maybe it shouldn't?
+        env_var_list <- c(env_var_list, setNames("BUNDLED", toupper(env_var)))
+      }
+    }
+  }
+
   env_var_list <- with_cloud_support(env_var_list)
 
   # turn_off_all_optional_features() needs to happen after
   # with_cloud_support(), since it might turn features ON.
+  # TODO: could we, should we have download_ok also include "download has succeeded"
+  # so we can disable these things if someone's machine is suddenly offline but they
+  # find themselves here and demand a working compilation?
```

Comment:
Though it should be noted that we _always_ need a modern cmake even if we are skipping all of the optional dependencies, so this won't give us a fully offline build in that case (old cmake + no internet)

Reply:
> So this would really just be for cases where the internet goes down mid run.

It also could happen on a CRAN system which has old cmake + is offline and is attempting to check. If that does happen, we should ask about upgrading cmake to modern versions (like is documented in their docs is how it is supposed to be).

Reply:
```suggestion
```
With cmake now in the sys reqs this comment can be removed.